### PR TITLE
Fix isort config

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
`isort` produces different results across Python version and environments.

**Description of the Change:**
Fix `isort` config and use GHA for isort and black. 

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/workflows/format.yml
Comment (AmintorDusko) [inline .github/workflows/format.yml:32]: We usually avoid third-party actions, but I guess we can rely on that.

File: Makefile
Comment (AmintorDusko) [inline Makefile:116]: So with this, we can fix the default to follow python3.11 version, right?

**Review by AmintorDusko — APPROVED** [review]
Hey @vincentmr, I assume you tested it with Python 3.9 (supported by us) and observed no more changes coming from it, right? In this case, thank you very much for that. Now that we have a consistent development environment, I'm happy. I will approve, but please wait for #646 to be merged, before merging this PR.

File: .github/workflows/format.yml
Comment (vincentmr) [reply .github/workflows/format.yml:32]: Yes, but they are the official isort/black collab actions so I thought it would be ok.

File: Makefile
Comment (vincentmr) [reply Makefile:116]: We needed all those options before things stabilized. I think the main issue was actually `-p pennylane_lightning` because sometimes it would be recognize as first party and others third party.

Comment (AmintorDusko) [reply Makefile:116]: Interesting. Thanks!

File: .github/workflows/format.yml
Comment (AmintorDusko) [reply .github/workflows/format.yml:32]: No problems for me.

**Review by maliasadi — CHANGES_REQUESTED** [review]
@vincentmr What about using these custom options in `pyproject.toml` so that we can reuse them in our GH action. Look at what we've done in Catalyst [for an example](https://github.com/PennyLaneAI/catalyst/blob/main/pyproject.toml).

Let's not use 3rd-part actions in our recipes where we can simply install these packages from PyPI ;)

**Comment by vincentmr** [comment]
> @vincentmr What about using these custom options in `pyproject.toml` so that we can reuse them in our GH action. Look at what we've done in Catalyst [for an example](https://github.com/PennyLaneAI/catalyst/blob/main/pyproject.toml).

Since this is only for `isort` at this point, I think the CLI is still doing the job.

> Let's not use 3rd-part actions in our recipes where we can simply install these packages from PyPI ;)

I would agree with the exception when the action is from the package collab, as in the present case.

File: .github/CHANGELOG.md
Comment (vincentmr) [reply .github/CHANGELOG.md:38]: I think it still is, I saw a case where `isort` will `black -l` is more opiniated than `black` itself.

Comment (vincentmr) [reply .github/CHANGELOG.md:38]: I would therefore keep it for good measure.

File: .github/workflows/format.yml
Comment (rashidnhm) [inline .github/workflows/format.yml:18]:
```suggestion
      - uses: actions/checkout@v4
```
Let's drop usage of `checkout@v3` across all our workflows as it uses a deprecated version of NodeJS

File: .github/workflows/format.yml
Comment (rashidnhm) [inline .github/workflows/format.yml:21]: If the action is provided by a trustworthy party, then *it should be* safe for use. I did look through the source code of the action and it seems simple enough, and does what we were doing previously with cli call

File: .github/workflows/format.yml
Comment (rashidnhm) [inline .github/workflows/format.yml:26]:
```suggestion
          requirements-files: 'requirements-dev.txt'
          sort-paths: './pennylane_lightning/ ./mpitests ./tests'
```

**Review by rashidnhm — COMMENTED** [review]
Would it make sense to use `make format-python` within CI instead of the actions? One benefit of doing that is it would ensure we're running identical code across both CI and local development environments.

**Comment by vincentmr** [comment]
> Would it make sense to use `make format-python` within CI instead of the actions? One benefit of doing that is it would ensure we're running identical code across both CI and local development environments.

Do you mean `make format-python check=1`?

File: Makefile
Comment (maliasadi) [reply Makefile:118]: chatted offline.. let's keep it for consistency

**Review by rashidnhm — APPROVED** [review]

**Pull request merged**
